### DISCUSSION TITLE
auth.resend() doesn't re-send the OTP via SMS if the user is already verified

### DISCUSSION THREAD

**Discussion by fbele** [supabase/supabase#21049] [Questions] [answered]
Hi all,

I am trying to implement a login flow using SMS channe with auth.signInWithOtp() method. The whole flow works as expected, even if the user doesn't exist yet, it gets created.
But if a user exists already and it was also verified some time in the past, the auth.resend() doesn't re-send the OTP via SMS. It does however work, if the user is newly created or hasn't been verified yet.
I am using Twilio Verify as my SMS provider.

Does anybody know how to resend an OTP, even if the user is already created and verified? Or is that a bug?
That's basically only thing missing/not working yet in my flow.

P.S.: If I click on "Send OTP" in my Users Dashboard, I also receive a new OTP, even for an already verified user. It only doesn't work through the API.

**Comment by GaryAustin1** [comment, marked as answer]
The dashboard is using signInWithOtp.

Resend is only used if there was an Otp already sent for the methods listed.

![image](https://github.com/supabase/supabase/assets/54564956/3f8dff14-254e-45f4-8177-06a7edac6aa7)

**Reply by fbele** [reply]
I see. Thank you for your reply.
Then, if I just use `signInWithOtp()` instead of `resend()` for both cases, there couldn't be anything really wrong, could it?
There is little to no difference then between `resend()` and `signInWithOtp()` the way I see it.

**Reply by GaryAustin1** [reply]
Resend is used if something goes wrong  with those specific operations and the user did not receive their email/message.

**Reply by fbele** [reply]
I understand, however, my thought was, that something could go wrong and the user did not receive their email/message also when the user is already existing and was verified. In such case there is no option to use `resend()` for user to get the email/message, as it only works when it's a new user logging in.

**Reply by GaryAustin1** [reply]
Correct if just signing in then you just use signInWithOtp again.  Or that is how I read it, resend won't work for that.

**Comment by fbele** [comment]
The signInWithOtp() method works as expected and I am using it to resend the OTP for existing and non-existing users.
